src/field/field3d.cxx

```diff
@@ -857,21 +882,26 @@ BoutReal mean(const Field3D &f, bool allpe, REGION rgn) {
 
   checkData(f);
 
-  // use first element for sum of values of f, second element for number of points
-  BoutReal result[2] = {0., 0.};
-  
-  for(const auto& i: f.region(rgn)) {
-    result[0] += f[i];
-    result[1] += 1.;
+  // Intitialise the cummulative sum and counter
+  BoutReal result = 0.;
+  int count = 0;
+
+  const Region<Ind3D> &region = f.getMesh()->getRegion3D(REGION_STRING(rgn));
+
+  BOUT_FOR_OMP(i, region, parallel for reduction(+:result,count)) {
+    result += f[i];
+    count += 1;
   }
 
   if(allpe) {
     // MPI reduce
-    BoutReal localresult[2] = {result[0], result[1]};
-    MPI_Allreduce(&localresult, &result, 2, MPI_DOUBLE, MPI_SUM, BoutComm::get());
+    BoutReal localresult = result;
+    MPI_Allreduce(&localresult, &result, 1, MPI_DOUBLE, MPI_SUM, BoutComm::get());
+    int localcount = count;
+    MPI_Allreduce(&localcount, &count, 1, MPI_INT, MPI_SUM, BoutComm::get());
```

Comment:
Yes I didn't enjoy doubling this but the `OpenMP` reduction clause wouldn't work with the original two element array block (should be supported in new versions of `OpenMP`, >= 4.5 I think). We could of course use `int` in the reduction and then pack it into a two element `BoutReal` array to be communicated to avoid this. I'd guess that the global synchronisation imposed by the first call might not be such a big cost for the second call as procs will already be close to in sync, but this is just a guess.